service/src/main/java/bio/terra/pipelines/app/controller/PipelineRunsApiController.java

```diff
@@ -190,13 +192,19 @@ public ResponseEntity<ApiGetPipelineRunsResponse> getAllPipelineRuns(
     PageResponse<List<PipelineRun>> pageResults =
         pipelineRunsService.findPipelineRunsPaginated(maxLimit, pageToken, userId);
 
+    // convert list of pipelines to map of id to name
+    Map<Long, String> pipelineIdToNameMap =
```

Comment:
just so i understand, you do this here so that you only need to query the pipelines table once to get this mapping that you can use to populate the pipelineNames, rather than needing to query it for each response?

Reply:
exactly! a little overkill when we have one pipeline but should be futureproof